### PR TITLE
Feature/df 489 stelle bewirtschaften backend

### PR DESCRIPTION
**Pull request by wwebec**

### CODE REVIEW

File: infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java [hunk 1]
```diff
@@ -140,15 +154,51 @@ public JobAdvertisementSearchService(CurrentUserContext currentUserContext,
...
+            ManagedJobAdSearchRequest searchRequest,
+            Pageable pageable) {
 
-        return new NativeSearchQueryBuilder()
-                .withFilter(filter);
+        SearchQuery query = new NativeSearchQueryBuilder()
+                .withQuery(createManagedJobAdsKeywordsQuery(searchRequest.getKeywords()))
+                .withFilter(mustAll(
+                        publicationStartDateFilter(searchRequest.getOnlineSinceDays()),
+                        ownerUserIdFilter(searchRequest.getOwnerUserId())))
```
Comment (cjpsycos) [inline infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java]: I think the companyIdFilter is missing

File: infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/ManagedJobAdSearchRequest.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+public class ManagedJobAdSearchRequest {
+
+    @Min(7)
+    @Max(365)
+    private Integer onlineSinceDays;
+
+    @NotBlank
+    private String ownerUserId;
+
```
Comment (cjpsycos) [inline infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/ManagedJobAdSearchRequest.java]: where is the ownerCompanyId  gone?

File: infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/ManagedJobAdSearchRequest.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    private String[] keywords;
```
Comment (cjpsycos) [inline infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/ManagedJobAdSearchRequest.java]: I don't think we receive an array, but just a string

File: infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java [hunk 4]
```diff
@@ -161,14 +211,12 @@ private BoolQueryBuilder titleFilter(PeaJobAdvertisementSearchRequest searchRequ
         return query;
     }
 
-    private BoolQueryBuilder publicationStartDatePeaFilter(PeaJobAdvertisementSearchRequest searchRequest) {
-        BoolQueryBuilder query = boolQuery();
-        if (searchRequest.getOnlineSinceDays() == null) {
-            return query;
+    private BoolQueryBuilder publicationStartDateFilter(Integer onlineSinceDays) {
```
Comment (cjpsycos) [inline infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java]: Actually it's the same as startDateFilter

File: infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/ManagedJobAdSearchRequest.java (hunk 2 above)
Comment (wwebec) [reply infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/ManagedJobAdSearchRequest.java]: according to the specification manged search request does not have filter nor field parameter ```ownerCompanyId```. It is a part of [```PeaJobAdvertisementSearchRequest```](https://github.com/alv-ch/job-ad-service/blob/master/infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/PeaJobAdvertisementSearchRequest.java)

File: infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java (hunk 1 above)
Comment (wwebec) [reply infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java]: ccording to the specification manged search request does not have filter nor field parameter ownerCompanyId,  please see [searchPeaJobAdvertisements@JobAdvertisementSearchService](https://github.com/alv-ch/job-ad-service/blob/93d4b139c429a10400d8d97c87355b88a36eb389/infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java#L148)

File: infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/ManagedJobAdSearchRequest.java (hunk 2 above)
Comment (cjpsycos) [reply infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/ManagedJobAdSearchRequest.java]: So how do you solve the PreAuthorize ...?
And what do you get back from the search request without the filter? The user can must only see the the JobAds of his company.

File: infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java (hunk 4 above)
Comment (wwebec) [reply infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java]: Could be more specific what do mean? This code is indeed the start date filter, beside name and small syntax nothing has been changed here.

File: infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java (hunk 1 above)
Comment (wwebec) [reply infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java]: companyIdFilter added

File: infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/ManagedJobAdSearchRequest.java (hunk 2 above)
Comment (wwebec) [reply infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/ManagedJobAdSearchRequest.java]: companyIdFilter added

File: infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java [hunk 5]
```diff
@@ -140,35 +156,64 @@ public JobAdvertisementSearchService(CurrentUserContext currentUserContext,
...
+                        ownerUserIdFilter(searchRequest.getOwnerUserId()),
+                        ownerCompanyIdFilter(searchRequest.getCompanyId())))
+                .withPageable(
+                        PageRequest.of(
+                                pageable.getPageNumber(),
+                                pageable.getPageSize(),
+                                Sort.by(desc(PATH_PUBLICATION_START_DATE))))
+                .build();
+
+        return jobAdvertisementElasticsearchRepository.search(query)
```
Comment (birom) [inline infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java]: Should not we implement highlighting?

Comment (cjpsycos) [reply infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java]: is coming in a second step. first goal is to enable the portal development

Comment (wwebec) [reply infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java]: added search highlighting of all fields

Comment (birom) [reply infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java]: Simply adding ``.withHighlightFields(new HighlightBuilder.Field("*"))`` is not going t work,  but  I think this portal team can start using this feature and we can implement the highlighting in a second PR.

**Pull request merged**